### PR TITLE
Feature/141126947/feedback

### PR DESCRIPTION
**Pull request by olawalejarvis**
### What does this PR does?
It implemented all the feedback gotten from Adebayo

### Description of task to be completed
- Make endpoint documentation the first section in Readme file
- Make each response message clear and readable
- Use appropriate HTTP status codes
- Protect Admin and Regular role from modification or deletion
- Break down user stories on pivotal tracker
- A user should not be able to update userId - send a response message explaining that a user can't change their id
- Return a message when a deleted user want to create a document
- Add pagination to get all documents and users methods
- Set a default limit and offset
- User should be able to search through their own documents
- Remove credentials from test in database config.json file
- Rename route's file names - instead of UserRouter use - User since route is the folder name
- Rename controller's file names - instead of UserController use User since controller is the folder name
- Separate controller's helper from the controller folder
Return right errors - return what is necessary for error messages
- All '.send()' response methods should start from a new line
- Add schema.sql file to root directory
- Remove roleId from the user's response data  
- Correct all line of codes that exceeded the maximum line length of 80
- Add spacing between test specifications
- Use middleware for all authentications
- Use HTTP 500 status code accurately
- Add logout functionality

### How should this be manually tested?
Run the following commands;
`git clone https://github.com/andela-oaladeusi/dms.git`
`cd dms`
`npm install`
`npm start`

### What are the relevant pivotal tracker stories?
https://www.pivotaltracker.com/story/show/141126947

### CODE REVIEW

File: .env.sample [hunk 1]
```diff
@@ -1,4 +1,6 @@
-PORT="" 
+PORT=""
 SECRET=""
 HOST=""
-DATABASE_URL=""
\ No newline at end of file
+DATABASE_URL=""
+DATABASE_DEV_URL=""
```
Comment (vonvick) [inline .env.sample]: `It's bad practice to have multiple environment variable declarations for the same resource like you have here: DATABASE_TEST_URL & DATABASE_DEV_URL you should have just one which is influenced by the environment running the application at that time. That's the whole point behind environment variables.`
-- from adebayo

File: server/app/Helper/Helper.js [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+      page_size: condition.limit,
+      total_count: condition.count
+    };
+  },
+  /**
+   * Get user's profile'
+   * @param {Object} data object containing user's details
+   * @returns {Object} return user's data
+   */
+  getUserProfile(data) {
```
Comment (vonvick) [inline server/app/Helper/Helper.js:59]: this method looks similar to the getUserAttributes method

File: server/tests/api/roles.spec.js [hunk 3]
```diff
@@ -39,41 +40,49 @@ describe('ROLE API', () => {
...
+            .equal('This field cannot be empty');
           done();
         });
     });
-    it('should return varification failed when no token is supplied', (done) => {
+
+    it('should return varification failed when no token is supplied',
+    (done) => {
       superRequest.post('/roles')
         .send(helper.testRoleG)
```
Comment (vonvick) [inline server/tests/api/roles.spec.js]: how about a more descriptive variable naming convention?

File: server/tests/model/document.spec.js [hunk 4]
```diff
@@ -11,7 +11,7 @@ describe('Document Model', () => {
   const emptyFields = ['title', 'content', 'access'];
 
   before((done) => {
-    db.Role.create(helper.testRoleR)
+    db.Role.create({ title: 'regular', id: 2 })
```
Comment (vonvick) [inline server/tests/model/document.spec.js:14]: Is there a need for id`?` postgres or mysql automatically assigns `id` to newly created items

File: .env.sample (hunk 1 above)
Comment (olawalejarvis) [reply .env.sample]: Okay
Thanks

File: server/app/Helper/Helper.js (hunk 2 above)
Comment (olawalejarvis) [reply server/app/Helper/Helper.js:59]: Yes
But, getUserAttributes method only returns fields key alone unlike getUserProfile  method that returns user's data

File: server/tests/api/roles.spec.js (hunk 3 above)
Comment (olawalejarvis) [reply server/tests/api/roles.spec.js]: Okay
Thanks

File: server/tests/model/document.spec.js (hunk 4 above)
Comment (olawalejarvis) [reply server/tests/model/document.spec.js:14]: Alright
Thanks

**Pull request merged**